lesson_6/lesson_6_homework.py

```diff
@@ -0,0 +1,133 @@
+    ('HR', ['61 ', '66 ', '65 ', '68 ', '77 ', '77 ', '76 ', '82 ', '81 ']),
+    ('HU', ['63 ', '67 ', '70 ', '73 ', '76 ', '79 ', '82 ', '83 ', '86 ']),
+    ('IE', ['78 ', '81 ', '82 ', '82 ', '85 ', '87 ', '88 ', '89 ', '91 ']),
+    ('IS', ['93 ', '95 ', '96 ', '96 ', ': ', ': ', '98 ', '99 ', '98 ']),
+    ('IT', ['62 ', '63 ', '69 ', '73 ', '75 ', '79 ', '81 ', '84 ', '85 ']),
+    ('LT', ['60 ', '60 ', '65 ', '66 ', '68 ', '72 ', '75 ', '78 ', '82 ']),
+    ('LU', ['91 ', '93 ', '94 ', '96 ', '97 ', '97 ', '97 ', '93 b', '95 ']),
+    ('LV', ['64 ', '69 ', '72 ', '73 ', '76 ', '77 b', '79 ', '82 ', '85 ']),
+    ('ME', [': ', '55 ', ': ', ': ', ': ', ': ', '71 ', '72 ', '74 ']),
+    ('MK', [': ', '58 ', '65 ', '68 ', '69 ', '75 ', '74 ', '79 ', '82 ']),
+    ('MT', ['75 ', '77 ', '78 ', '80 ', '81 ', '81 ', '85 ', '84 ', '86 ']),
+    ('NL', ['94 ', '94 ', '95 ', '96 ', '96 ', '97 ', '98 ', '98 ', '98 ']),
+    ('NO', ['92 ', '93 ', '94 ', '93 ', '97 ', '97 ', '97 ', '96 ', '98 ']),
+    ('PL', ['67 ', '70 ', '72 ', '75 ', '76 ', '80 ', '82 ', '84 ', '87 ']),
+    ('PT', ['58 ', '61 ', '62 ', '65 ', '70 ', '74 ', '77 ', '79 ', '81 ']),
+    ('RO', ['47 ', '54 ', '58 ', '61 ', '68 ', '72 ', '76 ', '81 ', '84 ']),
+    ('RS', [': ', ': ', ': ', ': ', '64 ', ': ', '68 ', '73 ', '80 ']),
+    ('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94 b', '95 ', '93 ', '96 ']),
+    ('SI', ['73 ', '74 ', '76 ', '77 ', '78 ', '78 ', '82 ', '87 ', '89 ']),
+    ('SK', ['71 ', '75 ', '78 ', '78 ', '79 ', '81 ', '81 ', '81 ', '82 ']),
+    ('TR', [': ', '47 ', '49 ', '60 ', '70 ', '76 ', '81 ', '84 ', '88 ']),
+    ('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
+    ('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 ']),
+]
+
+def get_dataset_as_dict(my_description, my_data):
+    year_list = [years for years in my_description[1]]
+    dataset = dict()
+    for element in my_data:
+        coverage_list = [coverage for coverage in element[1]]
+        country_year_coverage = list()
+        country_index = 0
+        for coverage in coverage_list:
+            if coverage != ': ' :
+                country_year_coverage.append({'year':year_list[country_index].replace(' ', ''), 'coverage':coverage.replace(' ', '')})   
+            country_index = country_index + 1
+
+        dataset[element[0]] = country_year_coverage
+
+    return dataset
+
+#end of get_dataset_as_dict function
+
+def get_year_dataset(my_dataset, year):
+    values_list = list()
+    for item in my_dataset: 
+         #print(item, my_dataset[item])
+         for dict_item in my_dataset[item]:
+            #print(dict_item)              
+            #print(dict_item['year'])
+            #print(dict_item['coverage'])
+            if dict_item['year'] == year:
+                values_list.append((item, dict_item['coverage']))
+                      
+                 
+    dataset_by_year = dict()
```

Comment:
daca vrei sa pastrezi datele petru fiecare an dupa mai multe apeluri ale functiei `get_year_dataset` este bine sa initializezi dictionarul dataset_by_year in exteriorul functiei. Daca pastram dictioanrul in interiorul functiei ii rescriem valoarea la fiecare apel si nu il putem accesa din exterior